homework/index.js

```diff
@@ -30,15 +32,97 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+  function renderContributors(contributors, contributorsContainer) {
+    createAndAppend('p', contributorsContainer, { text: 'Contributors', id: 'contributors' });
+    contributors.forEach(contributor => {
+      const contributorsDetailsDiv = createAndAppend('div', contributorsContainer, {
+        class: 'contributor-details',
+      });
+      createAndAppend('img', contributorsDetailsDiv, {
+        src: contributor.avatar_url,
+        alt: contributor.login,
+        class: 'image',
+      });
+      createAndAppend('a', contributorsDetailsDiv, {
+        text: contributor.login,
+        href: contributor.html_url,
+        target: '_blank',
+        class: 'contributor-data',
+      });
+      createAndAppend('p', contributorsDetailsDiv, {
+        text: contributor.contributions,
+        class: 'contributor-badge',
+      });
+    });
+  }
+
+  function fetchAndRenderData(selectedRepo, repoContainer, contributorsContainer, root) {
+    repoContainer.innerHTML = '';
+    contributorsContainer.innerHTML = '';
+
+    const updatedAt = new Date(selectedRepo.updated_at);
+    createAndAppend('p', repoContainer, {
+      text: `Repository: ${selectedRepo.name}`,
+      class: 'repo-child',
+    });
```

Comment:
This comment from week 1 is still left unimplemented:

> The assignment asks that the repo name is a link that opens the current GitHub repo in a new browser tab.

Reply:
Hi @remarcmij, thanks for the feedbacks and sorry I have not responded to the week1 feedback or even implemented the changes. I will try to do so this latest today. And will certainly improve on the CSS in the week3.

Reply:
Hi @remarcmij, all the changes have been made. The rest of the CSS will be handled in week3.